tests/parsers/test_dos.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=unused-argument
+"""Tests for the `DosParser`."""
+from __future__ import absolute_import
+import pytest
+from aiida.orm import FolderData
+from aiida.common import AttributeDict
+
+
+@pytest.fixture
+def dos_inputs():
+    # The DosParser doesn't really need to access the parent folder, but we'd like to make the inputs as realistic
+    # as possible, so we create an empty FolderData and attach it as an input to the current CalcJobNode.
+    inputs = {
+        'parent_folder': FolderData().store(),
+    }
+    
+    return AttributeDict(inputs)
+
+
+def test_dos_default(fixture_database, fixture_computer_localhost, generate_calc_job_node, generate_parser,
+                     dos_inputs, data_regression, num_regression):
+    """
+    Test `DosParser` on the results of a simple `dos.x` calculation.
+    """
+    entry_point_calc_job = 'quantumespresso.dos'
+    entry_point_parser = 'quantumespresso.dos'
+
+    node = generate_calc_job_node(entry_point_calc_job, fixture_computer_localhost, 'default', dos_inputs)
+    parser = generate_parser(entry_point_parser)
+    results, calcfunction = parser.parse_from_node(node, store_provenance=False)
+    
+    out_params = results['output_parameters'].get_dict()
+    out_dos_x = results['output_dos'].get_x()
+    out_dos_y = results['output_dos'].get_y()
+    dos_labels = [out_dos_x[0]] + [arr[0] for arr in out_dos_y]  # 4 strings
+    dos_values = [out_dos_x[1]] + [arr[1] for arr in out_dos_y]  # 4 numpy arrays
+    dos_units  = [out_dos_x[2]] + [arr[2] for arr in out_dos_y]  # 4 strings
+
+    assert calcfunction.is_finished, calcfunction.exception
+    assert calcfunction.is_finished_ok, calcfunction.exit_message
+    assert 'output_parameters' in results
+    assert 'output_dos' in results
+    data_regression.check({
+        'parameters': out_params,
+        'dos': {
+            'labels': dos_labels,
+            'units': dos_units,
+        }
+    })
+    num_regression.check({
+        'dos_val_{}'.format(i) : val for i,val in enumerate(dos_values)
+    }, default_tolerance=dict(atol=0, rtol=1e-18))
```

Comment:
Do these tolerances make sense? The magnitudes of the values range from 83.0 to 7.77e-88, which is why I set the absolute tolerance to zero. But I'm not sure about the relative tolerance.
Docs: https://pytest-regressions.readthedocs.io/en/latest/api.html#num-regression

Reply:
Is there a reason to suspect that the parsed values will change significantly? Did you try to use the normal `data_regression` on the `output_dos` node? Did that not work due to float precision?

Reply:
`data_regression` does not work on numpy arrays, as they are not yaml-serializable:
```
[...]
../../../venvs/aiida_py3.6/lib/python3.6/site-packages/yaml/representer.py:57: in represent_data
    node = self.yaml_representers[None](self, data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pytest_regressions.data_regression.RegressionYamlDumper object at 0x7fbc7cda0240>, data = array([-83.947, -83.937, -83.927, ...,  36.543,  36.553,  36.563])

    def represent_undefined(self, data):
>       raise RepresenterError("cannot represent an object: %s" % data)
E       yaml.representer.RepresenterError: cannot represent an object: [-83.947 -83.937 -83.927 ...  36.543  36.553  36.563]

../../../venvs/aiida_py3.6/lib/python3.6/site-packages/yaml/representer.py:229: RepresenterError
```

Reply:
We could deserialize them ourselves easily, before calling the `data-regression` no?

Reply:
I guess we could, but why?

Reply:
Let me clarify: if we serialize these numpy arrays ourselves, we still have to solve the problem of truncation and that of comparison tolerances. If all you want is to have zero tolerance, we can just set `atol=0` and `rtol=0` (it works on my machine, altough I suspect we might indeed incur in precision problems on different machines).

By the way, this is the comparison formula (from `numpy.isclose`): `absolute(a - b) <= (atol + rtol * absolute(b))`

Reply:
Fair enough. Honestly I don't have a good answer about the tolerance, but that is also not the focus of what we are testing here, so I think this is fine